### ISSUE TITLE
[BUG 2.15.0-beta] Closure actions can't receive primitive numbers

### ISSUE THREAD

**Issue by cibernox** [emberjs/ember.js#15470]
I found that you can't pass decimal or negative numbers to closure actions anymore:

Reproductions:
Twiddle: https://ember-twiddle.com/e255ab664fff5204791ed86577f0d7a1?openFiles=templates.application.hbs%2C
Other repo: https://github.com/cibernox/demo-error-closure-action

**Comment by machty** [comment]
👍 , This is breaking ember-concurrency builds testing against beta https://travis-ci.org/machty/ember-concurrency/jobs/251041405

**Comment by machty** [comment]
Specifically the ember-concurrency breakage is due to the following ([from here](http://localhost:4207/#/docs/examples/increment-buttons)):

```
<p>
  {{#press-and-hold-button
    press=(perform incrementBy -1)
    release=(cancel-all incrementBy)}}
      --Decrease
  {{/press-and-hold-button}}

  {{#press-and-hold-button
    press=(perform incrementBy 1)
    release=(cancel-all incrementBy)}}
      Increase++
  {{/press-and-hold-button}}
</p>
```

The raw `1` and `-1` are the primitives causing the breakage. (The `(perform)` helper just casts a task to a closure action.)

**Comment by ef4** [comment]
This also breaks liquid-fire's test suite. If anybody wants an easy test case, run the dummy app and visit /#/scenarios/model-dependent-rule/1

**Comment by rwjblue** [comment]
Thanks to @bekzod and @ef4's work over in glimmer-vm, https://github.com/emberjs/ember.js/issues/15572 has been submitted as a fix for this issue.

**Comment by rwjblue** [comment]
Can someone confirm that the latest canary builds (which will be published in ~ 15 minutes) fox the issue? I'll cherry pick the update into beta branch (and release a new beta) once confirmed...

**Comment by ef4** [comment]
I confirmed that canary fixes both liquid-fire's test suite and [Miguel's reproduction](https://github.com/cibernox/demo-error-closure-action).

**Comment by rwjblue** [comment]
Awesome, thank you for confirming @ef4! 

I just pulled the fix into `beta` branch and published a build.  I've also tagged and started publishing v2.15.0-beta.3 for those that are using this from npm.

I'm closing this for now, but I'd really appreciate it if everyone affected by this bug could test latest beta builds and confirm things are working properly now.  There is only one week before 2.15.0 final, and I'd really like to ensure we don't release with this issue...
